File: apps/sdkwork-mcp-flutter-mobile/packages/sdkwork_mcp_flutter_mobile_core/lib/bootstrap/sdk_clients.dart
import 'package:sdkwork_mcp_app_sdk_generated_flutter/app_client.dart';

import '../src/mcp_app_client.dart';
import '../src/session/mcp_app_session.dart';

class SdkworkMcpFlutterSdkClients {
  const SdkworkMcpFlutterSdkClients({
    required this.apiBaseUrl,
    required this.backendApiBaseUrl,
    required this.sdkFamilies,
    required this.appClient,
  });

  final String apiBaseUrl;
  final String backendApiBaseUrl;
  final List<String> sdkFamilies;
  final SdkworkAppClient appClient;
}

SdkworkMcpFlutterSdkClients createSdkworkMcpFlutterSdkClients({
  required String apiBaseUrl,
  String? backendApiBaseUrl,
  McpAppSession? session,
}) {
  final normalizedApiBaseUrl = apiBaseUrl.replaceAll(RegExp(r'/+$'), '');
  final bundle = createMcpAppClient(
    apiBaseUrl: normalizedApiBaseUrl,
    accessToken: session?.accessToken,
    authToken: session?.authToken,
    tenantId: session?.tenantId,
    organizationId: session?.organizationId,
    userId: session?.userId,
  );
  return SdkworkMcpFlutterSdkClients(
    apiBaseUrl: normalizedApiBaseUrl,
    backendApiBaseUrl: backendApiBaseUrl ??
        normalizedApiBaseUrl.replaceAll('/app/v3/api', '/backend/v3/api'),
    sdkFamilies: const [
      'sdkwork-mcp-app-sdk',
    ],
    appClient: bundle.appClient,
  );
}

File: apps/sdkwork-mcp-h5/src/bootstrap/sdkClients.ts
import {
  createMCPAppSdkClientConfig,
  getMCPAppSdkClient,
  initMCPAppSdkClient,
  type SdkworkMCPAppClient,
} from "@sdkwork/mcp-h5-core/sdk/mcpAppSdkClient";
import { getSdkworkChatGlobalTokenManager } from "@sdkwork/mcp-h5-core/session";

import { resolveEnvironment } from "./environment";

export interface SdkworkMcpH5SdkClientInventory {
  apiBaseUrl: string;
  backendApiBaseUrl: string;
  app: SdkworkMCPAppClient;
  sdkFamilies: {
    app: string[];
  };
}

export function bootstrapSdkClients(): SdkworkMcpH5SdkClientInventory {
  const environment = resolveEnvironment();
  getSdkworkChatGlobalTokenManager();
  initMCPAppSdkClient(createMCPAppSdkClientConfig());

  return {
    apiBaseUrl: environment.apiBaseUrl,
    backendApiBaseUrl: environment.backendApiBaseUrl,
    app: getMCPAppSdkClient(),
    sdkFamilies: {
      app: ["sdkwork-mcp-app-sdk"],
    },
  };
}
